packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -3646,6 +3652,45 @@ describeWithDOM('mount', () => {
         expect(wrapper.prop('bar')).to.equal('bye');
       });
     });
+
+    describe('props in async handler', () => {
+      class TestComponent extends React.Component {
+        constructor(props) {
+          super(props);
+          this.state = { counter: 1 };
+          this.handleClick = this.handleClick.bind(this);
+        }
+
+        handleClick() {
+          return delay(100).then(() => {
+            return new Promise((resolve) => {
+              this.setState({ counter: 2 }, () => {
+                resolve();
+              });
+            });
+          });
+        }
+
+        render() {
+          return (
+            <div id="parentDiv" onClick={this.handleClick}>
+              <TestSubComponent id="childDiv" counter={this.state.counter} />
+            </div>
+          );
+        }
+      }
+
+      class TestSubComponent extends React.Component {
+        render() {
+          return <div>{this.props.counter}</div>;
+        }
+      }
+      it('child component props should update after call to setState in async handler', () => {
+        const component = mount(<TestComponent />);
+        component.find('#parentDiv').props().onClick();
+        expect(component.find(TestSubComponent).props()).to.eql({ id: 'childDiv', counter: 2 });
```

Comment:
so, i would expect this test to fail - because the `setState` has `delay(100)` on it.

It's impossible for the test to know when that async action is completed, unless it hardcodes a `delay(100)`, or unless the `onClick` returns a promise.

Reply:
Yes. It will fail. I can work on trying to fix the bug too. Does the test look right though?

Reply:
No, to clarify - the test itself can't possibly pass. It's not possible for a test to magically wait for all async actions to be completed.

Reply:
Ah okay. That makes sense. I was working on it last night and am trying to figure out.

I thought of setting a
‘Const promise = component.find(‘parentDiv’).props().onClick()’
And then promise.resolve with the expect in it. But it doesnt seem to work. I was thinking that it might be a reasonable approach because the handleClick returns a promise. 

Other than that, I assume that having a delay(100) is not the best way to approach this problem. 

What’s your thought?

Reply:
I'm starting to think there's not actually a bug here, and that the original issue is invalid.

Reply:
ok actually, the original issue in #1872 is still a bug - i think the test case needs a bit of tweaking to more closely match that (without using await or async functions)

Reply:
I'm sorry but how should I go around writing this test? Or more, how am I able to know that I'm writing the test correctly?